plugins/terminal.koplugin/main.lua

```diff
@@ -1,58 +1,378 @@
+local ButtonDialog = require("ui/widget/buttondialog")
+local CenterContainer = require("ui/widget/container/centercontainer")
 local DataStorage = require("datastorage")
 local Font = require("ui/font")
 local InfoMessage = require("ui/widget/infomessage")
 local InputDialog = require("ui/widget/inputdialog")
+local LuaSettings = require("luasettings")
+local Menu = require("ui/widget/menu")
 local TextViewer = require("ui/widget/textviewer")
 local Trapper = require("ui/trapper")
 local UIManager = require("ui/uimanager")
 local WidgetContainer = require("ui/widget/container/widgetcontainer")
 local logger = require("logger")
 local util = require("ffi/util")
 local _ = require("gettext")
+local lfs = require("libs/libkoreader-lfs")
 local Screen = require("device").screen
 
-local Terminal = WidgetContainer:new{
+local Terminal = WidgetContainer:new {
     name = "terminal",
     dump_file = util.realpath(DataStorage:getDataDir()) .. "/terminal_output.txt",
     command = "",
+    items_per_page = G_reader_settings:readSetting("items_per_page") or 16,
+    source = 'terminal',
+    shortcuts_dialog = nil,
+    shortcuts_menu = nil,
+    shortcuts_file = DataStorage:getSettingsDir() .. "/terminal_shortcuts.lua",
+    shortcuts = {},
 }
 
 function Terminal:init()
     self.ui.menu:registerToMainMenu(self)
+    self:createSettingsFileIfNotExists()
+    local file_handle = LuaSettings:open(self.shortcuts_file)
+    self.shortcuts = file_handle:readSetting("shortcuts") or {}
+end
+
+function Terminal:createSettingsFileIfNotExists()
```

Comment:
This function seems either redundant or a bug in LuaSettings. Just doing a quick glance atm so I don't know which, although redundant seems more likely after years of using LuaSettings. ;-)